toqito/measurements/pretty_good_measurement.py

```diff
@@ -0,0 +1,67 @@
+"""Compute the set of pretty good measurements from an ensemble."""
+import numpy as np
+import scipy
+
+from toqito.matrix_ops import vector_to_density_matrix
+
+
+def pretty_good_measurement(states: list[np.ndarray], probs: list[float] | None = None) -> list[np.ndarray]:
+    r"""Return the set of pretty good measurements from a set of vectors and corresponding probabilities.
+
+    This computes the "pretty good measurement" as initially defined in :cite:`Hughston_1993_Complete`.
+
+    See Also
+    ========
+    pretty_bad_measurement
+
+    Examples
+    ========
+    Consider the collection of trine states.
+
+    .. math::
+        u_1 = |0\rangle, \quad
+        u_1 = -\frac{1}{2}\left(|0\rangle + \sqrt{3}|1\rangle\right), \quad \text{and} \quad
+        u_2 = -\frac{1}{2}\left(|0\rangle - \sqrt{3}|1\rangle\right).
+
+    >>> from toqito.states import trine
+    >>> from toqito.measurements import pretty_good_measurement
+    >>>
+    >>> states = trine()
+    >>> probs = [1 / 3, 1 / 3, 1 / 3]
+    >>> pgm = pretty_good_measurement(states, probs)
+    >>> pgm
+    [array([[0.66666667, 0.        ],
+           [0.        , 0.        ]]), array([[0.16666667, 0.28867513],
+           [0.28867513, 0.5       ]]), array([[ 0.16666667, -0.28867513],
+           [-0.28867513,  0.5       ]])]
+
+    References
+    ==========
+        .. bibliography::
+            :filter: docname in docnames
+
+
+    :raises ValueError: If number of vectors does not match number of probabilities.
+    :raises ValueError: If probabilities do not sum to 1.
+    :param states: A collection of either states provided as either vectors or density matrices.
+    :param probs: A set of probabilities.
+
+    """
+    n = len(states)
+
+    # If not probabilities are explicitly given, assume a uniform distribution.
+    if probs is None:
+        probs = n * [1 / n]
+
+    if len(states) != len(probs):
+        raise ValueError(f"Number of states {len(states)} must be equal to number of probabilities {len(probs)}")
+
+    if not np.isclose(sum(probs), 1):
+        raise ValueError("Probability vector should sum to 1.")
+
+    states = [vector_to_density_matrix(state) for state in states]
```

Comment:
```suggestion
    states = [vector_to_density_matrix(state) for state in states]
    # note that `vector_to_density_matrix` will return a density matrix if the provided input is a `rho-ensemble`. 
    # the function will alter the input only when the input ensemble is a set of vectors. 
```
This line confused me a bit for the case when the ensemble is a bunch of density matrices. So, I added these comments. Feel free to reject the addition.

Reply:
I might leave this out only because the docs for the function describe how it operates, so I feel like duplicating the description of the application of the function may be redundant.